tests/cloner_test.go

```diff
@@ -531,6 +531,77 @@ var _ = Describe("all clone tests", func() {
 			})
 			ClonerBehavior(cloneStorageClassName, "csivolumeclone")
 		})
+
+		Context("CloneStrategy on storageclass annotation", func() {
+			cloneType := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyCsiClone)
+
+			BeforeEach(func() {
+				if !f.IsCSIVolumeCloneStorageClassAvailable() {
+					Skip("CSI Volume Clone is not applicable")
+				}
+				cloneStorageClassName = f.CsiCloneSCName
+				By(fmt.Sprintf("Get original storage profile: %s", cloneStorageClassName))
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+				storageclass, err := f.K8sClient.StorageV1().StorageClasses().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				if storageclass.GetAnnotations() == nil {
+					storageclass.SetAnnotations(make(map[string]string))
+				}
+
+				storageclass.Annotations["cdi.kubevirt.io/clone-strategy"] = string(cloneType)
+				_, err = f.K8sClient.StorageV1().StorageClasses().Update(context.TODO(), storageclass, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() cdiv1.CDICloneStrategy {
+					storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					if storageProfile.Status.CloneStrategy == nil {
+						return ""
+					}
+					return *storageProfile.Status.CloneStrategy
+				}, time.Minute, time.Second).Should(Equal(cloneType))
+
+			})
+			AfterEach(func() {
+				By("[AfterEach] Restore the storage class - remove annotation ")
+				storageclass, err := f.K8sClient.StorageV1().StorageClasses().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				if storageclass.GetAnnotations() != nil {
+					delete(storageclass.Annotations, "cdi.kubevirt.io/clone-strategy")
+				}
+				_, err = f.K8sClient.StorageV1().StorageClasses().Update(context.TODO(), storageclass, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() cdiv1.CDICloneStrategy {
+					storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
```

Comment:
```
					if storageProfile.Status.CloneStrategy == nil {
						return ""
					}
```

Reply:
actually this way is correct
```
				Eventually(func() *cdiv1.CDICloneStrategy {
					storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
					Expect(err).ToNot(HaveOccurred())
					return storageProfile.Status.CloneStrategy
				}, time.Minute, time.Second).Should(BeNil())
```

Reply:
just rechecking does Eventually( expect cdiv1.CDICloneStrategy  with '*' ? or is it the otherway around? because i think it needs to be same with the eventually at BeforeEach() ?

Reply:
`storageProfile.Status.CloneStrategy` is a pointer, and the Eventually func wants the type. So you have dereference it with *storageProfile.Status.CloneStrategy. But before you dereference you have to make sure it is not nil.

Reply:
but I have also changed this -> now eventually wants `*cdiv1.CDICloneStrategy `

Reply:
but we can do it either way, just need to correct return type, the assertion and the return with nil check